pkg/handlers/generic/lifecycle/clusterautoscaler/handler.go

```diff
@@ -43,8 +44,9 @@ func (c *Config) AddFlags(prefix string, flags *pflag.FlagSet) {
 }
 
 type DefaultClusterAutoscaler struct {
-	client ctrlclient.Client
-	config *Config
+	client                ctrlclient.Client
+	config                *Config
+	helmAddonConfigGetter *config.HelmConfig
```

Comment:
Have you considered putting the getter in `helmAddonConfig`?

Reply:
Im not sure what you mean by this.